File: pY/edgeaipipeline_test.py
# edge_ai_pipeline_test.py
import unittest

class EdgeModel:
    def predict(self, x):
        return "active" if x > 0.5 else "inactive"

class TestEdgePipeline(unittest.TestCase):
    def test_inference(self):
        model = EdgeModel()
        self.assertEqual(model.predict(0.9), "active")
        self.assertEqual(model.predict(0.1), "inactive")

if __name__ == "__main__":
    unittest.main()

File: R/edgeaipipeline_test.R
# edge_ai_pipeline_test.R
library(testthat)

predict_edge <- function(x) {
  if (x > 0.5) "active" else "inactive"
}

test_that("Edge inference works", {
  expect_equal(predict_edge(0.9), "active")
  expect_equal(predict_edge(0.1), "inactive")
})
